src/gluonnlp/embedding/evaluation.py

```diff
@@ -0,0 +1,258 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=eval-used, redefined-outer-name
+"""Functions for intrinsic and extrinsic word embedding evaluation"""
+
+import mxnet as mx
+import numpy as np
+
+__all__ = ['CosineSimilarity', 'ThreeCosMul', 'ThreeCosAdd', 'HyperbolicCosineSimilarity']
+
+def _np_batch_dot(x, y):
+    return np.einsum("ij, ij -> i", x, y)
+
+def CosineSimilarity(x, y, eps=1e-10):
+    """Computes the cosine similarity.
+
+    Parameters
+    ----------
+    x : NDArray
+    y : NDArray
+    eps : float, optional, default=1e-10
+        A small constant for numerical stability.
+
+    Returns
+    -------
+    similarity : numpy.ndarray
+        The similarity scores between each word in x and each word in y.
+
+    """
+    dim = x.shape[-1]
+    assert dim == y.shape[-1], "Embedding dim for x, y are different!"
+    x = x.reshape(-1, dim)
+    y = y.reshape(-1, dim)
+    x = mx.nd.L2Normalization(x, eps=eps).asnumpy()
+    y = mx.nd.L2Normalization(y, eps=eps).asnumpy()
```

Comment:
How about to move this function to `op.py` and then reuse this function? https://github.com/dmlc/gluon-nlp/blob/a646c34304c4bde9423468714bd2ff6357cd2091/src/gluonnlp/attention_cell.py#L303-L318